src/details/ArborX_DetailsLegacy.hpp

```diff
@@ -55,6 +62,15 @@ class LegacyValues
   size_type size() const { return Access::size(_primitives); }
 };
 
+#if !defined(__NVCC__) ||                                                      \
+    (__CUDACC_VER_MAJOR__ * 100 + __CUDACC_VER_MINOR__ * 10) > 1103
+// FIXME: NVCC 11.0 has a bug with deduction guide here
+// Switch to using the deduction guide once we require higher version
+template <typename Primitives, typename BoundingVolume>
+LegacyValues(Primitives const &, BoundingVolume const &)
+    -> LegacyValues<Primitives, BoundingVolume>;
+#endif
```

Comment:
Do you need them at all? Shouldn't CTAD work automatically (https://godbolt.org/z/ar15xYx6P)?

Reply:
I had trouble with it. Some may have worked, but it was too fragile.

Reply:
Here's the [build](https://cloud.cees.ornl.gov/jenkins-ci/blue/organizations/jenkins/ArborX/detail/PR-915/7/pipeline) that failed on CUDA 11.0.3 without the guard:
```
src/ArborX_LinearBVH.hpp:143:49: error: 'LegacyValues' was not declared in this scope; did you mean 'ArborX::Details::LegacyValues'?
  143 |                           SpaceFillingCurve const &curve = SpaceFillingCurve())
      |                                                 ^~~~~~~~~~~~
      |                                                 ArborX::Details::LegacyValues
/var/jenkins/workspace/ArborX_PR-915/src/details/ArborX_DetailsLegacy.hpp:25:7: note: 'ArborX::Details::LegacyValues' declared here
```
The error made no sense, and was even pointing at the wrong place.

Reply:
What do you see if you don't provide the deduction guide at all?

Reply:
> What do you see if you don't provide the deduction guide at all?

Well, nothing, of course. It's not being used at the moment. I thought I would leave it there as a reminder of an action to take after we move away from 11.0.